stefaniaPopescu26/assignment3/Dictionary.java

```diff
@@ -65,60 +65,48 @@ private void add(String word) {
 
         TrieNode node = root;
 
-         do {
-            char letter = word.charAt(0);
+         for (int i = 0; i < word.length(); i++) {
+            char letter = word.charAt(i);
 
             if (!node.children.containsKey(letter)) {
                 node.children.put(letter, new TrieNode());
             }
 
             node = node.children.get(letter);
-            word = word.substring(1);
-        } while (!word.isEmpty());
+        }
 
          node.isWord = true;
     }
 
-    public boolean isPrefix(String prefix) {
-        if (prefix.isEmpty()) {
+    private boolean search(String word, String type) {
```

Comment:
The string type has only two values. Use a boolean that will check for "isPrefix" or "isWord" rather than a value of a string.